tinygrad/tensor.py

```diff
@@ -3354,6 +3357,10 @@ def dropout(self, p=0.5) -> Tensor:
     if not Tensor.training or p == 0: return self
     return (Tensor.rand_like(self, requires_grad=False, dtype=dtypes.default_float, contiguous=False) >= p).contiguous().where(self, 0) / (1.0 - p)
 
+  # helper function commonly used for indexing
+  def _one_hot_along_dim(self:Tensor, num_classes:sint, dim:int=0):
```

Comment:
it is weird that `dim` here is not the `dim` we used in other place. given that dim here is almost always 0, maybe we have the final reshape outside of this function

Reply:
Yeah that dim is weird. Kinda hacked just for getitem.

I don't think having reshape outside this function works since some use-cases requires the arange to target a specific dimension for masking

I tried to write it in a way where `dim` is still the `dim` concept used in other places. So instead of `0` as default, it's `-1` to indicate we're masking the `self` Tensor on the last dim.

Reply:
okay, will merge this in first. it's definitely a good thing to factor this idea into a function